client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -140,6 +140,30 @@ export async function getAllCategoriesAndCounts(
 
   return Object.fromEntries(arrayOfLabelsAndCounts);
 }
+export async function getAllCategories(
+  category: string,
+  page: Page
+): Promise<string[]> {
+  // these load asynchronously, so we have to wait for the specific category.
+  const categoryRows = await page
+    .getByTestId(`category-${category}`)
+    .getByTestId("categorical-row")
+    .all();
+
+  const arrayOfLabels = await Promise.all(
+    categoryRows.map(async (row): Promise<string> => {
+      const cat = await row
+        .getByTestId("categorical-value")
+        .getAttribute("aria-label");
+
+      if (!cat) throw new Error("category value not found");
+
+      return cat;
+    })
+  );
```

Comment:
discussion: Not sure how flaky this is. Is it worth adding a `tryUntil()` to ensure `arrayOfLabels` all resolve properly before returning it?

E.g.,

```ts
const categoryRows = await page
    .getByTestId(`category-${category}`)
    .getByTestId("categorical-row")
    .all();

// this retries until we get all the labels 
await tryUntil(async () => {
  await Promise.all(
    categoryRows.map(async (row): Promise<string> => {
      const cat = await row
        .getByTestId("categorical-value")
        .getAttribute("aria-label");

      if (!cat) throw new Error("category value not found");

      return cat;
    })
  );
}, { page })

const arrayOfLabels = await Promise.all(
    categoryRows.map(async (row): Promise<string> => {
      const cat = await row
        .getByTestId("categorical-value")
        .getAttribute("aria-label");

      if (!cat) throw new Error("category value not found");

      return cat;
    })
  );

return arrayOfLabels;
```

Reply:
I don't think it's flakey.